cmake/Modules/GrMinReq.cmake

```diff
@@ -0,0 +1,26 @@
+# Copyright 2011-2021 Free Software Foundation, Inc.
+#
+# This file is part of GNU Radio
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+if(DEFINED __INCLUDED_GR_MIN_REQ_CMAKE)
+    return()
+endif()
+set(__INCLUDED_GR_MIN_REQ_CMAKE TRUE)
+# Minimum dependency versions for central dependencies:
+set(GR_BOOST_MIN_VERSION "1.65")      ## Version in Ubuntu 18.04LTS
+set(GR_CMAKE_MIN_VERSION "3.10.2")    ## Version in Ubuntu 18.04LTS
+set(GR_MAKO_MIN_VERSION "1.0.7")      ## debian buster, 18.04LTS
+set(GR_PYTHON_MIN_VERSION "3.6.5")    ## Version in Ubuntu 18.04LTS
+set(GR_NUMPY_MIN_VERSION "1.13.3")    ## Version in Ubuntu 18.04LTS
+set(GR_PYGCCXML_MIN_VERSION "2.0.0")  ## Version to support c++17 (in pip)
+set(GCC_MIN_VERSION "8.3.0")          ## debian buster
+set(CLANG_MIN_VERSION "11.0.0")       ## debian bullseye, Fedora 33
+set(APPLECLANG_MIN_VERSION "1100")    ## same as clang 11.0.0, in Xcode11
+set(MSVC_MIN_VERSION "1914")          ## VS2017 15.7, for full-ish C++17 support
+set(VOLK_MIN_VERSION "2.4.1")         ## first version with CPU features
+set(PYBIND11_MIN_VERSION "2.4")       # pybind11 sets versions like 2.4.dev4, which compares < 2.4.3
+set(GR_THRIFT_MIN_VERSION "0.13")
```

Comment:
I recommend to stick with the versions that are set in the top-level `CMakeLists.txt`.
```
set(GR_BOOST_MIN_VERSION "1.69")      ## Version in CentOS 8 (EPEL)
set(GR_CMAKE_MIN_VERSION "3.16.3")    ## Version in Ubuntu 20.04LTS
set(GR_MAKO_MIN_VERSION "1.1.0")      ## Version in Ubuntu 20.04LTS
set(GR_NUMPY_MIN_VERSION "1.17.4")    ## Version in Ubuntu 20.04LTS
set(GCC_MIN_VERSION "9.3.0")          ## Version in Ubuntu 20.04LTS
```
Especially, GCC 8.3 will be annoying because it requires special treatment for `std::filesystem` while GCC 9+ does not. 
I'd argue that the next GR release doesn't need to support legacy systems that will become obsolete while GR 3.10 is still the current version.

For MSVC, I suggest to bump the minimum version to `1931` just to get rid of some headache with incompatibilities. Or at least `1920`, which is the first VS2019 release.